src/components/CertificateDropZone/CertificateDropZone.tsx

```diff
@@ -1,112 +1,88 @@
-  verificationError,
-  resetData,
-  toggleQrReaderVisible,
-  isDragAccept,
-  isDragReject,
-}) => {
-  // isDragReject is checking for mimetype (but we skipped it)
-  // fileError is when the file is not in JSON format and threw when deserilising
-  // valid JSON files will be handled by handleCertificateChange()
+  const isVerificationPending = verificationPending;
+  const isVerificationError = verificationStatus && !isValid(verificationStatus);
+  const isActionError = retrieveCertificateByActionState === states.FAILURE;
 
-  // change in prop does not update state, needs to update state
-  const [verificationErrorMessage, setVerificationError] = useState(verificationError);
-  useEffect(() => {
-    setVerificationError(verificationError);
-  }, [verificationError]);
+  const resetData = useCallback(() => {
+    dispatch(resetCertificateState());
+  }, [dispatch]);
 
-  if (isDragReject || fileError) {
-    return <DefaultView hover={true} accept={false} toggleQrReaderVisible={toggleQrReaderVisible} />;
-  }
-  if (isDragAccept) {
-    return <DefaultView hover={true} accept={true} toggleQrReaderVisible={toggleQrReaderVisible} />;
-  }
-  if (verificationError) {
-    return (
-      <DefaultView
-        hover={true}
-        accept={true}
-        toggleQrReaderVisible={toggleQrReaderVisible}
-        verificationError={verificationErrorMessage}
-      />
-    );
-  }
-  if (verifying) {
-    return <VerifyingView />;
-  }
-  if (!!retrieveCertificateByActionError) {
-    return <UnverifiedView resetData={resetData} retrieveCertificateByActionError={retrieveCertificateByActionError} />;
-  }
-  if (verificationStatus && !isValid(verificationStatus)) {
-    return <UnverifiedView verificationStatus={verificationStatus} resetData={resetData} />;
-  }
-  return <DefaultView hover={false} accept={true} toggleQrReaderVisible={toggleQrReaderVisible} />;
-};
+  const onDrop = useCallback(
+    (acceptedFiles: Blob[]) => {
+      acceptedFiles.forEach((file: Blob) => {
+        const reader = new FileReader();
 
-const onFileDrop = (
-  acceptedFiles: Blob[],
-  handleCertificateChange: (json: string) => void,
-  handleFileError: (error: any) => void
-) => {
-  const reader = new FileReader();
-  if (reader.error) {
-    handleFileError(reader.error as any);
-  }
-  reader.onload = () => {
-    try {
-      const json = JSON.parse(reader.result as string);
-      handleCertificateChange(json);
-    } catch (e) {
-      handleFileError(e);
-    }
-  };
-  if (acceptedFiles && acceptedFiles.length && acceptedFiles.length > 0)
-    acceptedFiles.map((f: Blob) => reader.readAsText(f));
-};
+        reader.onabort = () => console.log("file reading was aborted");
+        reader.onerror = () => console.log("file reading has failed");
+        reader.onload = () => {
+          try {
+            const json = JSON.parse(reader.result as string);
+            dispatch(updateCertificate(json));
+          } catch (e) {
+            console.error(e);
```

Comment:
in this case, if upload file has error, what will happen? since we are not handling the file error, can we add it into the log too?
i think if we enable the limitation of file extensions it will come into this catch error. should we handle cases like this?

Reply:
yeah once enable, will reach there. will want to handle such cases in upcoming pr